### DISCUSSION TITLE
After changing the transportation technology energy discount rate, the transportation services cost did not change

### DISCUSSION THREAD

**Discussion by barryliu54** [JGCRI/gcam-core#466] [Modifying GCAM]
Hi,
I tried modifying the energy discount rate for the transportation sector by altering the following line in the constants.R file in the GCAM\gcam-v7.1-Windows-Release-Package\input\gcamdata\R folder: 

energy.DISCOUNT_RATE_VEH <- 0.1 # Consumer discount rate for vehicle purchases 
to
energy.DISCOUNT_RATE_VEH <- 1 # Consumer discount rate for vehicle purchases 

However, the cost of transportation services did not change significantly.

I'm curious about how to interpret this situation? Does this mean my modification method is incorrect, or is it that the energy.DISCOUNT_RATE_VEH does not have a significant impact on the energy technology of the transportation sector?"

**Comment by pkyle** [comment]
This parameter, in constants.R, is only used for leveling capital costs of a portion of transportation technologies; I think right now it's just the light duty vehicles. Other transportation technologies have their costs already levelized when they are read into gcamdata, using the standard 10% discount rate. Still I would expect that this would increase the costs of the technologies of `trn_pass_road_LDV_4W` by a lot (~7x); can you confirm that these stayed the same?
I wouldn't expect this to change the prices of anything outside of the light duty vehicles, and also at the higher levels where the time value component is a large portion of the reported cost, the effect should be comparatively small as well.

**Reply by barryliu54** [reply]
Hello, Kyle,

Thank you so much! 

What a coincidence! I'm currently focused on the light duty transportation sector. However, after checking, I found that attempting to change `energy.DISCOUNT_RATE_VEH` does not seem to affect the costs of the technologies. 

One possible issue I thought of is whether modifying the parameter in the `constants.R` file requires running any files to compile the change other than just running `run-gcam.bat` for the changes to `energy.DISCOUNT_RATE_VEH` to take effect?

**Comment by pkyle** [comment]
Oh sorry for the misunderstanding--yes you'll need to re-build the XML files after making this change. Running gcam does not automatically re-build XML files. To do that you can key in `make xml` to the root directory of the workspace, from a command prompt or git bash window. Note that you'll need to have R installed, along with all necessary packages; the packages used are listed in the imports section of file `input/gcamdata/DESCRIPTION`. I don't see these XML build steps listed in the documentation but I wrote them down a few years ago here: https://github.com/JGCRI/gcam-core/issues/123#issuecomment-725148119
